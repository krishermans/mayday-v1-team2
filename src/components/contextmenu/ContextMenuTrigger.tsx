import * as React from "react";
import { forwardRef, ReactNode } from "react";

import type { MenuTriggerChildProps } from "@fluentui/react-components";


const ContextMenuTrigger = forwardRef<HTMLDivElement, Partial<MenuTriggerChildProps> & {
  children?: ReactNode
}>((props, ref) =>
  (
    <div {...props} ref={ref}>
      { props.children }
    </div>
  ));

ContextMenuTrigger.displayName = "ContextMenuTrigger";

export default ContextMenuTrigger;