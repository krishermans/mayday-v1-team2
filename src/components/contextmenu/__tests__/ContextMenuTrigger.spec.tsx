import React from "react";
import { render } from "@testing-library/react";
import ContextMenuTrigger from "../ContextMenuTrigger";

describe("ContextMenuTrigger Test Suite", () => {

    test("Renders correctly with children", () => {
        const { container } = render(<ContextMenuTrigger><h1>Test</h1></ContextMenuTrigger>);

        expect(container).toMatchSnapshot();
    });

});