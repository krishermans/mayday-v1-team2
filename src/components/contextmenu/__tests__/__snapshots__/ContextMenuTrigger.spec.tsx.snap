// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextMenuTrigger Test Suite Renders correctly with children 1`] = `
<div>
  <div>
    <h1>
      Test
    </h1>
  </div>
</div>
`;
